addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -0,0 +1,387 @@
+        """ "Check policy allows pre processing as not all polices
+        can be used in this way
+        """
+        func = getattr(self, "_get_suggested_location_" + policy, None)
+
+        if not hasattr(func, "_allow_preprocess"):
+            raise ValidationError(
+                _("This policy(%s) is not meant to be used in " "preprocessing") % policy
+            )
+    
+    def get_suggested_locations(self, move_line_ids, limit=None, sort=True):
+        """Dispatch the configured suggestion location policy to
+        retrieve the suggested locations
+        """
+        result = self.env["stock.location"]
+
+        # WS-MPS: use self.ensure_one() and don't loop (self is used in all but
+        # one place), or suggest locations for the move lines of the picking,
+        # use picking instead of self inside the loop and ?intersect? result.
+        for picking in self:
+            policy = picking.picking_type_id.u_drop_location_policy
+
+            if policy:
+                if (
+                    move_line_ids
+                    and picking.picking_type_id.u_drop_location_preprocess
+                    and not move_line_ids.any_destination_locations_default()
+                ):
+                    # The policy has been preprocessed this assumes the
+                    # the policy is able to provide a sensible value (this is
+                    # not the case for every policy)
+                    # Use the preselected value
+                    result = self._get_suggested_location_exactly_match_move_line(move_line_ids)
+
+                    # Just to prevent running it twice
+                    if not result and policy == "exactly_match_move_line":
+                        return result
+
+                # If the pre-selected value is blocked
+                if not result:
+                    func = getattr(self, "_get_suggested_location_" + policy, None)
+                    if func:
+                        # The signature of each "get suggested location" method may not
+                        # match, so inspect the method and only pass in the expected args
+                        expected_args = inspect.getfullargspec(func)[0]
+                        keys = ["move_line_ids", "limit", "sort"]
+                        local_vars = locals()
+                        kwargs = {x: local_vars[x] for x in keys if x in expected_args}
+                        result = func(**kwargs)
+                        # If result is sorted by the suggest method, don't re-sort it
+                        if "sort" in expected_args:
+                            sort = False
+        if sort:
+            return result.sorted(lambda l: l.name)
+        else:
+            return result
+    
+    def _get_suggested_location_exactly_match_move_line(self, move_line_ids):
+        self._check_picking_move_lines_suggest_location(move_line_ids)
+        location = move_line_ids.mapped("location_dest_id")
```

Comment:
Not sure: think that in version 14 is not needed .mapped for id-s. Needs to be tried first though.
```suggestion
        location = move_line_ids.location_dest_id
```

Reply:
This can be done in a few places :)